gtk_gui/welcome_window.py

```diff
@@ -0,0 +1,85 @@
+import gi
+
+from interfaces.welcome_window_interface import WelcomeWindowInterface
+
+gi.require_version('Gtk', '3.0')
+from gi.repository import Gdk, GObject, Gtk, GdkPixbuf
+
+
+class Handler:
+    @staticmethod
+    def on_destroy(*args):
+        Gtk.main_quit()
+
+
+class Components():
+    def __init__(self, builder):
+        self._allComponents = {}
+        self._builder = builder
+
+    def load_component_main_window(self, config):
+        for component_name in config["window_list_components"]:
+            self._allComponents[component_name] = self._builder.get_object(component_name)
+
+    def set_component(self, component_name, component):
+        self._allComponents[component_name] = component
+
+    def get_component(self, component_name):
+        return self._allComponents[component_name]
+
+
+@WelcomeWindowInterface.register
+class WelcomeWindowGtk(WelcomeWindowInterface):
+
+    def __init__(self, language_manager, config_general, config_main_window):
+        GObject.threads_init()
+        Gdk.threads_init()
+
+        self._settings = Gtk.Settings.get_default()
+        self._config_main_window = config_main_window
+        self._config_general = config_general
+        self._pages = list()
+        self._page_index = 0
+        self._lang_manager = language_manager
+        self._builder = Gtk.Builder()
+        self._builder.add_from_file(self._config_main_window["window_xml_file"])
+
+        self._builder.connect_signals(
+            {
+                'onDestroy': Handler.on_destroy,
+                'onButtonExit': Handler.on_destroy
+            })
+        self._component = Components(self._builder)
+
+        self.init_window()
+        self.set_full_screen_win()
```

Comment:
I think you should inline those functions instead of calling them.
There name suggests they are setters, but they aren't, and unless i'm mistaken  they are only called once here.